### ISSUE TITLE
Favor single quote's where possible

### ISSUE THREAD

**Issue by oliv3r** [koalaman/shellcheck#2346]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

I've been a 'double quote' guy for all my live, as C says, a string is surrounded by double quotes' ;) Single quotes are for characters! However in shell, this is not true. As most, I was stubborn and used double quotes whenever anyway ;)

After years, I've come to the realization, from a safety/security point of view, it actually does make sense to favor single quotes in scripts. It indicates 'no processing/expansion on the content of this string shall be done' (right?). As such, the choice (recommendation) should be, that whenever a variable does not contain anything that ought to be processed, favor single quotes.

Linters for example, such as pep8 also do this afaik as well.

**Comment by brother** [comment]
> As such, the choice (recommendation) should be, that whenever a variable does not contain
> anything that ought to be processed, favor single quotes.

(bash)
```shell
echo 'big up'
myvar='hello'

# rather than

echo "I know my BGP"
myothervar="world"

# and be careful with processing

echo "$myvar $myothervar"
echo '$myvar $myothervar'
```
Interesting to see if there are data from the big set @koalaman uses for testing that would indicate that this would be a good choice. For my own code I think I more often have processing involved and mostly need to avoid spacing issues and so on. But I haven't looked.

**Comment by koalaman** [comment]
This is a pretty subjective style debate that it doesn't make sense for ShellCheck to pick sides in, but it could potentially be an explicitly enabled optional suggestion.

**Comment by felipecrs** [comment]
+1 to be an optional rule, preferably with "Did you mean" suggestions so that it can be [auto-fixed with VS Code ShellCheck on save](https://github.com/vscode-shellcheck/vscode-shellcheck/pull/278).

**Comment by felipecrs** [comment]
Or:

```
enable=all
disable=favor-single-quotes
```

**Comment by oliv3r** [comment]
Ah yeah, that looks good to me too.

**Comment by felipecrs** [comment]
I changed my mind, I think this should be an optional rule just like `prefer-double-brackets`.

**Comment by oliv3r** [comment]
haha, but `all` enables all optionals no? or is it
```
enable=all, prefer-double-brackets
```
cause that's kind of weird :p

**Comment by ramses0** [comment]
(please disregard reference from #2687, it was a typo of 2346 v. 2364)

**Comment by ramses0** [comment]
Please see the linked issue #2687 ... I believe a strong case can be made that `'` should be preferred for as many strings as possible.  Similar preferences and reasoning can be found in other linters (eg: https://eslint.org/docs/latest/rules/quotes ):

"""Object option:

"avoidEscape": true allows strings to use single-quotes or double-quotes so long as the string contains a quote that would have to be escaped otherwise"""

**Comment by oliv3r** [comment]
BTW, we already have the opposite in SC2016 :)

**Comment by ramses0** [comment]
@oliv3r - is this then just a matter of "patches welcome"?

**Comment by oliv3r** [comment]
@ramses0 I don't know? That's up to the maintainer; but I would expect it to be so?
